bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java

```diff
@@ -0,0 +1,305 @@
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.openthermgateway.OpenThermGatewayBindingConstants;
+import org.openhab.binding.openthermgateway.internal.DataItem;
+import org.openhab.binding.openthermgateway.internal.DataItemGroup;
+import org.openhab.binding.openthermgateway.internal.GatewayCommand;
+import org.openhab.binding.openthermgateway.internal.GatewayCommandCode;
+import org.openhab.binding.openthermgateway.internal.LogLevel;
+import org.openhab.binding.openthermgateway.internal.Message;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewayCallback;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewayConfiguration;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewayConnector;
+import org.openhab.binding.openthermgateway.internal.OpenThermGatewaySocketConnector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OpenThermGatewayHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Arjen Korevaar
+ */
+@NonNullByDefault
+public class OpenThermGatewayHandler extends BaseThingHandler implements OpenThermGatewayCallback {
+
+    private final Logger logger = LoggerFactory.getLogger(OpenThermGatewayHandler.class);
+
+    @Nullable
+    private OpenThermGatewayConfiguration config;
+
+    @Nullable
+    private OpenThermGatewayConnector connector;
+
+    public OpenThermGatewayHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing OpenTherm Gateway handler for uid '{}'", getThing().getUID());
+
+        updateStatus(ThingStatus.OFFLINE);
+
+        config = getConfigAs(OpenThermGatewayConfiguration.class);
+
+        connect();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Received channel: {}, command: {}", channelUID, command);
+
+        try {
+            if (command.toFullString() != "REFRESH") {
+                String channel = channelUID.getId();
+                String code = getGatewayCodeFromChannel(channel);
+
+                GatewayCommand gatewayCommand;
+                if (command instanceof QuantityType) {
+                    gatewayCommand = GatewayCommand.parse(code,
+                            Double.toString(((QuantityType) command).doubleValue()));
+                } else {
+                    gatewayCommand = GatewayCommand.parse(code, command.toFullString());
+                }
+
+                if (gatewayCommand != null && checkConnection()) {
+                    connector.sendCommand(gatewayCommand);
+                }
+            }
+        } catch (Exception ex) {
+            logger.error("error", ex);
```

Comment:
Do not log to error. This is reserved for framework issues. What kind of errors do you expect? Do they need to be reported or is this merely debugging information?